src/test/scala/piconot/external/parser/parserTests.scala

```diff
@@ -0,0 +1,154 @@
+package piconot.external.parser
+
+import org.scalatest._
+import picolib.semantics._
+import piconot.external._
+
+/**
+  * Created by tmf on 10/23/16.
+  */
+class parserTests extends FunSuite with Matchers {
+  // some syntactic sugar for expressing parser tests
+  implicit class ParseResultChecker(input: String) {
+    def ~>(output: List[List[Rule]]) = {
+      val result = PiconotParser(input)
+      result.successful && result.get == output
+    }
+  }
+
```

Comment:
This is pretty similar to the tests we implemented. I wonder if there was an easy way to be able to test all of the cases, kind of like forall with integers.